impl/datasource-providers/GeniusYield/OrderBot/DataSource/Providers.hs

```diff
@@ -80,14 +64,14 @@ runQuery (Connection nid providers) = runGYTxQueryMonadNode nid providers
 
 allOrderInfos
     :: Connection
-    -> DEX
+    -> DEXInfo
     -> [OrderAssetPair]
     -> IO (Map OrderAssetPair [SomeOrderInfo])
 allOrderInfos c dex assetPairs = do
     cTime <- getCurrentGYTime
 
     partialOrderInfos <- runQuery c $
-                         runReaderT (partialOrders $ partialOrderFilter cTime) dex
+                         runReaderT (partialOrdersWithTransformerPredicate (dexPORefs dex) $ partialOrderFilter cTime) dex
```

Comment:
`partialOrdersWithTransformerPredicate` has type signature,
```hs
partialOrdersWithTransformerPredicate
  ∷ GYDexApiQueryMonad m a
  ⇒ PORefs
  → (PartialOrderInfo → Maybe b)
  → m [b]
```
whereas `partialOrders` have type signature,
```hs
partialOrders
  ∷ GYDexApiQueryMonad m a
  ⇒ PORefs
  → m (Map.Map GYTxOutRef PartialOrderInfo)
```
And definition of `partialOrdersWithTransformerPredicate` is
```hs
partialOrdersWithTransformerPredicate pors transformerPredicate = do
  ois ← Map.elems <$> partialOrders pors
  pure $ mapMaybe transformerPredicate ois
```